lib/DAV/WorkspacePlugin.php

```diff
@@ -93,24 +93,24 @@ public function propFind(PropFind $propFind, INode $node) {
 		if (!$workspaceAvailable || !$workspaceEnabled) {
 			return;
 		}
-
-		$propFind->handle(self::WORKSPACE_PROPERTY, function () use ($node) {
-			/** @var Folder[] $nodes */
-			$nodes = $this->rootFolder->getUserFolder($this->userId)->getById($node->getId());
-			if (count($nodes) > 0) {
-				/** @var File $file */
-				try {
-					$file = $this->workspaceService->getFile($nodes[0]);
-					if ($file instanceof File) {
-						return $file->getContent();
-					}
-				} catch (StorageNotAvailableException $e) {
+		if ($propFind->getDepth() > 0) {
```

Comment:
Isn't this actually the wrong logic for your issue description? I assumed if a depth > 1 it should just return and not handle the prop find.

Reply:
no, when depth is 1 or greater, it should return the rich workspace, meaning :
when depth = 1, we have actually looking at  the requesting folder , we should return the rich workspace 
when depth = 0, we are getting items within the requested folder(the 22 subfolders ), we do not need to return the rich workspace for them, the client can always request them separately

Reply:
Ah right, never mind. Thanks for the summary again.